test/auth.test.js

```diff
@@ -58,15 +67,48 @@ describe('Auth', () => {
     return client.login('email', 'password')
       .then(userId => expect(userId).toEqual('fake-user-id'));
   });
+});
 
-  it('should return a promise for login with only existing auth data userId', async() => {
-    window.fetch.resolves(mockApiResponse());
-    expect.assertions(1);
+describe('Auth login semantics', () => {
+  const email = 'test_user@domain.com';
+  const password = 'password';
+  let test = new StitchMongoFixture();
+  let th;
+  let client;
 
-    let client = new StitchClient();
-    await client.auth.storage.set(common.USER_AUTH_KEY, mockAuthData());
+  beforeAll(async() => await test.setup());
+  afterAll(async() => await test.teardown());
 
-    return client.login('email', 'password')
-      .then(userId => expect(userId).toEqual('fake-user-id'));
+  beforeEach(async() => {
+    const { apiKey, groupId, serverUrl } = extractTestFixtureDataPoints(test);
+    th = await buildClientTestHarness(apiKey, groupId, serverUrl);
+    await th.configureAnon();
+    client = th.stitchClient;
+  });
+
+  afterEach(async() => await th.cleanup());
+
+  it('should track currently logged in provider type', async() => {
```

Comment:
[minor] We should assert a logout did indeed happen besides just re logging in.

Reply:
I assert this in a later test.

Reply:
I don't see that the other test actually asserts that we sent a logout to the server instead of re logging in.

Reply:
Didn't assert, but it's checked implicitly in `should re-login with the same provider if not anon`. I'll add the assertion.